### PR TITLE
fix: 'Last Purchase Rate' taking wrong on BOM

### PR DESCRIPTION
**Pull request by lokesh-indictrans**
#20228 'Last Purchase Rate' taking wrong on BOM.:
2nd Commit:
While fetching the last purchase rate, when the purchase order and purchase receipt both are none, the '>=' condition becomes true and it throws fatal error.  
Added condition when both purchase order and purchase receipt are none.

**Issue** 
1st Commit: I created one purchase order for item 'A' and enter rate manually as 20.
Again i created a new purchase order for same item and entered rate as 25.
Then, I go to BOM, selected rate of materials based on as  'Last Purchase Rate'.
Select item 'A' rate gets fetch as 25.
Then i cancelled and deleted that purchase order of rate 25. 
Again go to BOM select rate of materials based on as  'Last Purchase Rate'.
Selected item 'A'. Still it fetches rate as 25,It should fetch rate as 20 because the purchase order with rate 25 has been deleted.
bom_issue
After resolving this output come as:
BOM_issue

### CODE REVIEW

**Comment by lokesh-indictrans** [comment]
Hi, As per your request i have send this PR again with only one commit and description.

**Review by nextchamp-saqib — APPROVED** [review]

File: erpnext/stock/doctype/item/item.py [hunk 1]
```diff
@@ -1002,14 +1004,17 @@ def get_last_purchase_details(item_code, doc_name=None, conversion_rate=1.0):
...
 
+	
+	
 	purchase_order_date = getdate(last_purchase_order and last_purchase_order[0].transaction_date
 							   or "1900-01-01")
 	purchase_receipt_date = getdate(last_purchase_receipt and
 								 last_purchase_receipt[0].posting_date or "1900-01-01")
 
-	if (purchase_order_date > purchase_receipt_date) or \
+	if (purchase_order_date >= purchase_receipt_date) or \
```
Comment (nextchamp-saqib) [inline erpnext/stock/doctype/item/item.py]: When `last_purchase_order` and `last_purchase_receipt` are both `None`, `purchase_order_date ` and `purchase_receipt_date ` becomes equal and the `>=` messes up tests.

change condition to `if last_purchase_order and (purchase_order_date >= purchase_receipt_date) or \.....`

**Review by nextchamp-saqib — DISMISSED** [review]

**Comment by lokesh-indictrans** [comment]
@nextchamp-saqib As per your suggestion, i have added that condition. Please check.

**Review by nabinhait — APPROVED** [review]

**Comment by marination** [comment]
@Mergyfio backport version-12-hotfix

**Comment by marination** [comment]
@Mergifyio backport version-12-hotfix

**Pull request merged**